frontend/src/pages/Mesh/target/TargetPanelCommon.tsx

```diff
@@ -184,6 +184,221 @@ export const renderNodeHeader = (
           </span>
         </>
       )}
+      {link}
     </React.Fragment>
   );
 };
+
+const badgeStyle = kialiStyle({
+  display: 'inline-block',
+  marginRight: '0.25rem',
+  marginBottom: '0.25rem'
+});
+
+const summaryStyle = kialiStyle({
+  marginBottom: '0.25rem'
+});
+
+const summaryHeaderStyle = kialiStyle({
+  marginLeft: '0.75rem'
+});
+
+const summaryInfoStyle = kialiStyle({
+  marginLeft: '2.0rem'
+});
+
+const infraStyle = kialiStyle({
+  marginTop: '0.75rem'
+});
+
+export const renderNodeLink = (meshData: MeshNodeData, style?: string): React.ReactNode | undefined => {
+  let displayName, key, link: string;
+  let pfBadge: PFBadgeType;
+
+  switch (meshData.infraType) {
+    case MeshInfraType.GATEWAY:
+      const name = `${meshData.infraName}-${meshData.infraData?.spec?.gatewayClassName}`;
+      link = `/namespaces/${encodeURIComponent(meshData.namespace)}/workloads/${encodeURIComponent(name)}`;
```

Comment:
If the gatewayClassName is undefined, it could lead to an incorrect link. This could happen if you create a Gateway instead of a k8s Gateway:
![image](https://github.com/user-attachments/assets/fed704c8-0af3-41af-a192-095533a936d9)
Probably could be better removing the last part of the name when it is undefined?

Reply:
good catch, I'll make the change.